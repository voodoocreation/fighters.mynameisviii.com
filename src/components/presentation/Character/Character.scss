@import "../../../scss/variables", "../../../scss/mixins";

.Character {
  position: relative;
  flex-basis: 100%;

  h2 {
    padding: $spacing;
    margin: 0;
  }

  h3 {
    margin-top: 0;
  }

  &--image {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background: transparent 50% no-repeat;
    background-size: cover;
    filter: blur(10px);
    opacity: 0.2;
    z-index: 0;
  }

  &--content {
    position: relative;
    z-index: 1;
  }

  &--moveset {
    padding: 0 $spacing $spacing-half;

    dl {
      margin: 0;
    }

    dd {
      margin: 0 0 $spacing-half;
    }

    + *::before {
      display: block;
      width: 30%;
      height: 1px;
      content: "";
      margin: 0 auto $spacing-half;
      background: linear-gradient(to right, rgba($text-color, 0), rgba($text-color, 0.7), rgba($text-color, 0));
    }

    &:last-child {
      padding-bottom: $spacing;
    }
  }

  @include screen-below($screen-xs-max) {
    &--content {
      position: absolute;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;
      overflow: auto;
    }
  }

  @include screen-above($screen-sm-min) {
    flex-basis: 50%;
    padding: $spacing-half;

    &--image {
      top: $spacing;
      right: $spacing;
      bottom: $spacing;
      left: $spacing;
      opacity: 0.6;
    }

    &--content {
      height: 100%;
      border: $transparent-border-color 1px solid;
      background: $panel-1-bg;
    }
  }
}

.Move--conditions {
  font-size: 0.8rem;
  font-weight: 400;
  font-style: italic;
}
