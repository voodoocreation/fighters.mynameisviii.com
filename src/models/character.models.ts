import { DeepPartial } from "redux";

import { createSlugFromString } from "../helpers/dataTransformers";
import { IMoveset, moveset } from "./moveset.models";

export interface ICharacter {
  imageUrl: string;
  movesets: IMoveset[];
  name: string;
  slug: string;
}

export const character = (
  options: DeepPartial<ICharacter> = {}
): ICharacter => ({
  imageUrl: options.imageUrl || "",
  movesets: options.movesets ? options.movesets.map(moveset) : [],
  name: options.name || "",
  slug: options.slug || createSlugFromString(options.name || "")
});
