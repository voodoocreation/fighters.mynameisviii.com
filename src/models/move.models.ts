import { DeepPartial } from "redux";

export interface IMove {
  combo: string;
  conditions?: string;
  name: string;
}

export const move = (options: DeepPartial<IMove> = {}): IMove => ({
  combo: options.combo || "",
  conditions: options.conditions || undefined,
  name: options.name || ""
});
