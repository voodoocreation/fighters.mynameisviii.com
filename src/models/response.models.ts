export interface ISuccess<Data> {
  ok: true;
  data: Data;
}

export const success = <T>(data: T): ISuccess<T> => ({ ok: true, data });

export interface IFailure {
  ok: false;
  message: string;
}

export const failure = (message: string): IFailure => ({ ok: false, message });
