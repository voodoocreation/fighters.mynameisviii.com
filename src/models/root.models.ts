export * from "./character.models";
export * from "./error.models";
export * from "./game.models";
export * from "./move.models";
export * from "./moveset.models";
export * from "./response.models";
