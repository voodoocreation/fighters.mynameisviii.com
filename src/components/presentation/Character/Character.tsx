import * as React from "react";
import { InjectedIntlProps, injectIntl } from "react-intl";

import { ICharacter, IMoveset } from "../../../models/root.models";

import Combo from "../Combo/Combo";

import "./Character.scss";

interface IProps extends ICharacter, InjectedIntlProps {}

class Character extends React.Component<IProps> {
  public render() {
    const { imageUrl, name, movesets, slug } = this.props;

    return (
      <article className={`Character Character--${slug}`}>
        <div
          className="Character--image"
          style={{ backgroundImage: `url(${imageUrl})` }}
        />

        <div className="Character--content">
          <h2 id={slug}>{name}</h2>

          {movesets.map(this.renderMoveset)}
        </div>
      </article>
    );
  }

  private renderMoveset = (moveset: IMoveset) => (
    <section className="Character--moveset" key={moveset.name}>
      <h3>{moveset.name}</h3>

      <dl>
        {moveset.moves.map(move => (
          <React.Fragment key={move.name}>
            <dl>
              <span>{move.name}</span>{" "}
              {move.conditions ? (
                <span className="Move--conditions">{move.conditions}</span>
              ) : null}
            </dl>

            <dd className="Move--combo">
              <Combo>{move.combo}</Combo>
            </dd>
          </React.Fragment>
        ))}
      </dl>
    </section>
  );
}

const CharacterWrapped = injectIntl(Character);

export default CharacterWrapped;
