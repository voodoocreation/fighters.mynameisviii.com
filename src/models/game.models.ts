import { DeepPartial } from "redux";

import { createSlugFromString } from "../helpers/dataTransformers";
import { character, ICharacter } from "./character.models";

export interface IGame {
  imageUrl: string;
  slug: string;
  title: string;
  characters: ICharacter[];
}

export const game = (options: DeepPartial<IGame> = {}): IGame => ({
  characters: options.characters ? options.characters.map(character) : [],
  imageUrl: options.imageUrl || "",
  slug: options.slug || createSlugFromString(options.title || ""),
  title: options.title || ""
});
