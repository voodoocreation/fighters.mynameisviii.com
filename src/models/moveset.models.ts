import { DeepPartial } from "redux";

import { IMove, move } from "./move.models";

export interface IMoveset {
  moves: IMove[];
  name: string;
}

export const moveset = (options: DeepPartial<IMoveset> = {}): IMoveset => ({
  moves: options.moves ? options.moves.map(move) : [],
  name: options.name || "",
});
