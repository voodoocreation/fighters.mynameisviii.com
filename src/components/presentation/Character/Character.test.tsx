import { character, move, moveset } from "../../../models/root.models";
import ComponentTester from "../../../utilities/ComponentTester";

import Character from "./Character";

const testCharacter = character({
  imageUrl: "imageUrl",
  movesets: [
    moveset({
      moves: [
        move({
          combo: "D,F,LP",
          conditions: "conditions",
          name: "Move 1"
        })
      ],
      name: "Moveset with conditions"
    }),
    moveset({
      moves: [
        move({
          combo: "D,F,LP",
          name: "Move 2"
        })
      ],
      name: "Moveset without conditions"
    })
  ],
  name: "Character Name"
});

const component = new ComponentTester(Character).withDefaultProps(
  testCharacter
);

describe("[presentation] <Character />", () => {
  const { actual } = component.render();

  it("renders the correct id on the heading", () => {
    expect(actual.find("h2").attr("id")).toBe("character-name");
  });

  it("renders all movesets", () => {
    expect(actual.find(".Character--moveset")).toHaveLength(
      testCharacter.movesets.length
    );
  });

  it("only renders the conditions for the moves that have them", () => {
    expect(actual.find(".Move--conditions")).toHaveLength(1);
  });

  it("matches snapshot", () => {
    expect(actual).toMatchSnapshot();
  });
});
