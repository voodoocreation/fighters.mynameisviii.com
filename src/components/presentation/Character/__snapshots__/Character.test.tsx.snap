// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <Character /> matches snapshot 1`] = `
<article
  class="Character Character--character-name"
>
  <div
    class="Character--image"
    style="background-image:url(imageUrl)"
  />
  <div
    class="Character--content"
  >
    <h2
      id="character-name"
    >
      Character Name
    </h2>
    <section
      class="Character--moveset"
    >
      <h3>
        Moveset with conditions
      </h3>
      <dl>
        <dl>
          <span>
            Move 1
          </span>
           
          <span
            class="Move--conditions"
          >
            conditions
          </span>
        </dl>
        <dd
          class="Move--combo"
        >
          <div
            class="Combo"
          >
            <span
              class="ComboButton ComboButton--d"
            >
              <svg
                class="ComboButton--icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
                />
              </svg>
            </span>
            <span
              class="ComboButton ComboButton--f"
            >
              <svg
                class="ComboButton--icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
                />
              </svg>
            </span>
            <span
              class="ComboButton ComboButton--lp"
            >
              <span
                class="ComboButton--text"
              >
                LP
              </span>
            </span>
          </div>
        </dd>
      </dl>
    </section>
    <section
      class="Character--moveset"
    >
      <h3>
        Moveset without conditions
      </h3>
      <dl>
        <dl>
          <span>
            Move 2
          </span>
           
        </dl>
        <dd
          class="Move--combo"
        >
          <div
            class="Combo"
          >
            <span
              class="ComboButton ComboButton--d"
            >
              <svg
                class="ComboButton--icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
                />
              </svg>
            </span>
            <span
              class="ComboButton ComboButton--f"
            >
              <svg
                class="ComboButton--icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
                />
              </svg>
            </span>
            <span
              class="ComboButton ComboButton--lp"
            >
              <span
                class="ComboButton--text"
              >
                LP
              </span>
            </span>
          </div>
        </dd>
      </dl>
    </section>
  </div>
</article>
`;
